loader/loader.go

```diff
@@ -170,16 +171,80 @@ func (l *Loader) Load(patterns ...string) ([]*GunkPackage, error) {
 			// A Go package that isn't a Gunk package - skip it.
 			continue
 		}
+		if err := validatePackage(pkg); err != nil {
+			return nil, err
+		}
 		if l.cache == nil {
 			l.cache = make(map[string]*GunkPackage)
 		}
 		l.cache[pkg.PkgPath] = pkg
 		pkgs = append(pkgs, pkg)
 	}
-
 	return pkgs, nil
 }
 
+// validatePackage will validate that a given GunkPackage. It should
+// check sanity checks and common error cases that are shared with
+// the format and generate packages.
+func validatePackage(gunkPackage *GunkPackage) (formatErr error) {
```

Comment:
just a thought - perhaps this could return a `[]error`. it depends on how easy it is for these validity checks to continue after finding one error. not a problem either way, but it would be easier for the human if the tool gave many errors at once, just like most compilers do.

Reply:
Done.